### ISSUE TITLE
Unable to run sysdig container

### ISSUE THREAD

**Issue by luca3m** [coreos/bugs#2239]
# Issue Report #

With the latest CoreOS Alpha sysdig container isn't anymore able to compile the kernel module on the fly

## Bug ##

The reason looks like that `Makefile` inside `/lib/modules/<kernel version>/build` changed, now it points to a directory that does not exists:

```
DKMS make.log for sysdig-0.19.1 for kernel 4.14.0-rc8-coreos (x86_64)
Thu Nov  9 11:08:26 UTC 2017
make: Entering directory '/host/lib/modules/4.14.0-rc8-coreos/build'
make[1]: Entering directory '/host/lib/modules/4.14.0-rc8-coreos/build'
make[1]: *** /build/amd64-usr/var/tmp/portage/sys-kernel/coreos-modules-4.14_rc8/work/coreos-modules-4.14_rc8/source: No such file or directory.  Stop.
make[1]: Leaving directory '/host/lib/modules/4.14.0-rc8-coreos/build'
Makefile:24: recipe for target '__sub-make' failed
make: *** [__sub-make] Error 2
make: Leaving directory '/host/lib/modules/4.14.0-rc8-coreos/build'
root@8ca26469958a:/#
```

### Container Linux Version ###

```
core@ip-10-0-0-12 ~ $ cat /etc/os-release
NAME="Container Linux by CoreOS"
ID=coreos
VERSION=1590.0.0
VERSION_ID=1590.0.0
BUILD_ID=2017-11-08-0831
PRETTY_NAME="Container Linux by CoreOS 1590.0.0 (Ladybug)"
ANSI_COLOR="38;5;75"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://issues.coreos.com"
COREOS_BOARD="amd64-usr"
core@ip-10-0-0-12 ~ $
```

### Environment ###

Spawn a VM on Amazon EC2

### Expected Behavior ###

sysdig container being able to compile the kernel module and load it.

### Actual Behavior ###

Kernel module compilation fails

### Reproduction Steps ###

  1. Spawn a CoreOS EC2 instance
  2. run: 

```
docker run -i -t --rm --name sysdig --privileged -v /var/run/docker.sock:/host/var/run/docker.sock -v /dev:/host/dev -v /proc:/host/proc:ro -v /boot:/host/boot:ro -v /lib/modules:/host/lib/modules:ro -v /usr:/host/usr:ro sysdig/sysdig
```

To play with it you can also add an `--entrypoint=bash` and run `/docker-entrypoint.sh` manually so you can have a shell to look around.

**Comment by bgilbert** [comment]
Thanks for the report. 4.14 made some changes to path calculation in the kernel build system, and it seems coreos/linux@bee4c2c9f88e6f0ebb1281f939a7f97922fa494d is no longer doing what we want.

**Comment by bgilbert** [comment]
It turns out that our 4.14-rc kernel builds no longer include any Container Linux-specific patches. [`K_FROM_GIT=1`](https://github.com/coreos/coreos-overlay/blob/ba95a60415321cd884165e66749fb52884a7e801/sys-kernel/coreos-sources/coreos-sources-4.14_rc8.ebuild#L13-L14) has the side effect of [preventing subsequent patches from being applied](https://github.com/coreos/portage-stable/blob/d88bf167d48539323cec7397d206c0a07874097c/eclass/kernel-2.eclass#L1304).

This should be resolved on Monday when 4.14 final is released.

**Comment by bgilbert** [comment]
This should be fixed in 1590.1.0, due shortly.  Thanks for reporting.
